connectors/src/lib/temporal_monitoring.ts

```diff
@@ -114,8 +114,24 @@ export class ActivityInboundLogInterceptor
         if (connectorId) {
           await syncFailed(connectorId, "oauth_token_revoked");
 
-          this.logger.info("Cancelling workflow because of expired token.");
-          await cancelWorkflow(workflowId);
+          // Probes in the workflowId to identify long running workflows (such as Notion workflows
+          // which are expected to be never canceled).
+          const LONG_RUNNING_WORKFLOWS_PROBES: string[] = ["notion"];
```

Comment:
This does not feel robust, can't we have either the workflow type in the memo object or a flag like `cancelOnTokenRevoked` ?

It can be retrieved like that:
https://github.com/dust-tt/dust/blob/main/connectors/src/lib/temporal.ts#L76

And can be set here:

https://github.com/dust-tt/dust/blob/main/connectors/src/connectors/notion/temporal/client.ts#L50

Reply:
It's cleaner I agree but disperse the logic in many places. Which comes with advantages and disadvantages...

Reply:
I would think this is the direction we want to go to scale connectors:
Have the connectorId, connectorProvider and maybe even the connector workspaceId attached to each workflow to be able to do all the cleaning necessary when deleting a workspace, a connector, etc.

Reply:
I agree that relying on the name today is probably ok but this can quickly become something we forget about and it can backfire on us at any time.

Reply:
Maybe at least add a comment describing what could be a more long term solution?

Reply:
Happy to add `cancelOnTokenRevoked` :+1: We should also definitely have all that defined centrally for connectors of course.

This should be a base class option that we extend and that startWorkflow is standardize around that base class etc etc.. :)

Reply:
We definitely need a better structure and the right tooling around that.
If we feel like it's time to do it, I am happy to work on it in the upcoming days.